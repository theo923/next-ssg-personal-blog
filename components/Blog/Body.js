import React from "react";
import BlogList from "./BlogList";
import dynamic from "next/dynamic";
const BlogPost = dynamic(() => import("./BlogPost"));

const Body = ({ category, data, item, httpString }) => {
    return (
        <div data-test="component-blog-body" className="blogBody">
            <div className="blogBodyDiv">
                {httpString ? (
                    <BlogPost data={data} httpString={httpString} />
                ) : (
                    <BlogList category={category} data={data} item={item} />
                )}
            </div>
        </div>
    );
};

export default Body;
