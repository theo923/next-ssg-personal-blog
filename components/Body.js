import React from 'react';


const Body = ({item}) => {
    return (
        <div className=' rounded-2xl shadow-2xl mt-12 mx-4 mb-2 md:mx-8 md:mb-8 xl:m-12 ' style={{height: 3000}} >
            <div>{item}</div>
        </div>

    )
    
}

export default Body;
