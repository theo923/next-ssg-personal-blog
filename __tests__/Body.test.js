import { shallow } from 'enzyme';
import React from 'react';
import Body from '../components/Blog/Body';

describe('Body test', () => {

    it("expect Body component is rendered without crashing", () => {
        const wrapper = shallow(<Body />);   
    });

    it("expect Body component is rendered && match with the Snapshot", () => {
        const wrapper = shallow(<Body />);   
        expect(shallow(<Body />)).toMatchSnapshot();
    });
    
});

