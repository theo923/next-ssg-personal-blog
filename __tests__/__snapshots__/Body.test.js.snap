// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body test expect Body component is rendered && match with the Snapshot 1`] = `
<div
  className=" mt-12 mx-4 mb-2 md:mx-8 md:mb-8 "
>
  <div
    className=" flex flex-col"
    style={
      Object {
        "height": 3000,
      }
    }
  >
    <BlogList />
  </div>
</div>
`;
